### PR TITLE
Add timestamped streams

### PR DESCRIPTION
**Pull request by Gold872**
Added ability to stream a pair of the current value and timestamp, along with an option to always yield a value even if it's a duplicate (useful for graph widgets)

### CODE REVIEW

**Comment by mjansen4857** [comment]
A few suggestions:

I think this should use [records](https://dart.dev/language/records) instead of a pair. This could avoid needing to use another package and has the benefit of being able to name the fields. So for example you could access the timestamp via `returnValue.timestamp` instead of whatever the name of it in a pair would be. Or just separate out the values similar to how multiple returns work in python"
```var (value, timestamp) = methodThatReturnsARecord();```

Second, I think the timestamp should be stored and returned as a DateTime object, instead of an integer timestamp. This is just overall more useful. There is a `DateTime.fromMicrosecondsSinceEpoch` constructor that can be used to create one right from the timestamp given by the NT server.

File: lib/src/nt4_client.dart [hunk 1]
```diff
@@ -388,8 +419,20 @@ class NT4Client {
...
+        WebSocketChannel.connect(Uri.parse(serverAddr), protocols: [
+      'networktables.first.wpi.edu',
+      'v4.1.networktables.first.wpi.edu',
+    ]);
+
+    // Prevents connecting to the wrong address when changing IP address
+    if (!serverAddr.contains(serverBaseAddress)) {
+      return;
+    }
+
```
Comment (mjansen4857) [inline lib/src/nt4_client.dart]: Also, it looks like you have some duplicate code in here from the last PR. These lines for example. There's a few others throughout the changes as well

Comment (Gold872) [reply lib/src/nt4_client.dart]: I did the merging stuff wrong (gonna have to reset part of the branch) so all of the commits appear again

**Comment by Gold872** [comment]
I'll reopen a new one when I fix this merging stuff

**Pull request closed without merging**